### DISCUSSION TITLE
Validation rules are getting applied for hidden fields. How can I handle that.

### DISCUSSION THREAD

**Discussion by neumadhu** [colinhacks/zod#3298] [Q&A] [answered]
Based on the selection I am hiding/showing some of the form fields. But when I submit the form hidden fields validation is also triggering and I am not able to submit the form.  I have tried with ".partial()" also but still it's not working. Please let me know how we can resolve this issue.

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.

**Reply by robin-blanchard** [reply]
This [sandbox](https://codesandbox.io/p/devbox/keen-dream-nfqljj?workspaceId=ws_HtaXrLmagdQERUnia9UkeL) seems to reproduce the situation. I don't know if zod or React Hook Form is responsible for this
